lib/spear/connection/request.ex

```diff
@@ -0,0 +1,194 @@
+defmodule Spear.Connection.Request do
+  @moduledoc false
+
+  # a struct representing a stream-able request
+
+  @type t :: %{
+          continuation: Enumerable.continuation(),
+          request_ref: Mint.Types.request_ref(),
+          buffer: binary(),
+          from: GenServer.from(),
+          response: Spear.Connection.Response.t(),
+          status: :streaming | :done
+        }
+
+  defstruct [:continuation, :buffer, :request_ref, :from, :response, :status]
+
+  def new(event_stream, request_ref, from) do
+    reducer = &reduce_with_suspend/2
+    stream = Stream.map(event_stream, &Spear.Request.to_wire_data/1)
+    continuation = &Enumerable.reduce(stream, &1, reducer)
```

Comment:
this is where our enumerable magic begins

it's worth checking out this implementation in GenStage which reads `demand` (integer) number of elements from a stream in a similar fashion: https://github.com/elixir-lang/gen_stage/blob/6cb08ee2e892b8aff0fcc57f45e05c8d02dddcb2/lib/gen_stage/streamer.ex#L11-L27